platform/linux-generic/arch/aarch64/odp_sysinfo_parse.c

```diff
@@ -202,6 +202,49 @@ int cpuinfo_parser(FILE *file, system_info_t *sysinfo)
 
 void sys_info_print_arch(void)
 {
+	int value;
+
+	printf("ARM FEATURES:\n");
+	printf("  __ARM_ARCH           %i\n",  __ARM_ARCH);
+
+	value = -1;
+#ifdef __ARM_ARCH_ISA_A64
+	value = __ARM_ARCH_ISA_A64;
+#endif
+	printf("  __ARM_ARCH_ISA_A64   %i\n", value);
+
+#ifdef __ARM_FEATURE_QRDMX
+	/* Actually, this checks for new NEON instructions in
+	 * v8.1, but is currently the only way to distinguish
+	 * v8.0 and >=v8.1. */
+	printf("    ARMv8.1 instructions\n");
+#endif
+
+	value = -1;
+#ifdef __ARM_NEON
+	value = __ARM_NEON;
+#endif
+	printf("  __ARM_NEON           %i\n", value);
+
+	value = -1;
+#ifdef __ARM_FEATURE_IDIV
```

Comment:
I don't quite like this kind of code. Maybe we can use the following template?

```c
#ifdef __ARM_FEATURE
   printf("  __ARM_FEATURE", __ARM_FEATURE);
#endif
```

Reply:
I want to print out outcome of all features tested (so that it's easy to compare the output). Most feature flags are not defined when the feature is disabled, and define 1 when it's enabled. So, above example would not print anything when it's disabled.

Reply:
I'm ok with current changes.  while value var is not needed here.  And ifdef can go inside print. But that is tiny cosmetic thing. I mean like that:

```
	printf("SOMETHING %d\n",
#if defined(SOMETHING)
	       SOMETHING
#else
	       -1
#endif
	      );
```

Reply:
@muvarov please no. This is IMO the ugliest possible form.
Moreover if somebody uses `#define` to define printf, things will break, as preprocessors do not have to support directives inside macro arguments.

Reply:
@lumag  ok then looks like this PR can go in.